stl/inc/bitset

```diff
@@ -40,6 +40,36 @@ __declspec(noalias) bool __stdcall __std_bitset_from_string_2(void* _Dest, const
 #endif // _USE_STD_VECTOR_ALGORITHMS
 
 _STD_BEGIN
+template <class _Ty, size_t _Size, bool _Dynamic = (_Size * sizeof(_Ty) > _STD_BITSET_TO_STREAM_STACK_RESERVATION)>

```

Comment:
The `_Dynamic` criterion is improper.

The stack buffer is `_Ty _Buf[_STD_BITSET_TO_STREAM_STACK_RESERVATION];` and the dynamic option is `new _Ty[_Size];`. You should switch to dynamic when `_Size` exceeds the stack reservation constant, i.e. the `sizeof` shouldn't factor into the calculation. (If you wanted the constant to be a number of bytes, then the stack buffer's size is wrong. Element/byte confusion is dangerous, so proper unit naming is critical if you're crossing between domains.)

This was clearer to me after renaming the parameters.

Reply:
I wanted the constant stack size (consistent with `basic_string` SSO)

Reply:
Oh, I see now. What I intended was not complete. `_Bits` should have been passed there instead of `_STD_BITSET_TO_STREAM_STACK_RESERVATION`. I still think it is better to do that, than have constant element reservation.

Reply:
I mean like `_Elem _Buf[_Bits ? _Bits : 1];`